diff -ru src/hdf5-orig/src/H5Tvlen.c src/hdf5-2.0.0/src/H5Tvlen.c
--- src/hdf5-orig/src/H5Tvlen.c	2025-11-11 00:00:49.000000000 +0000
+++ src/hdf5-2.0.0/src/H5Tvlen.c	2025-12-04 18:18:18.262401600 +0000
@@ -1013,10 +1013,15 @@
             }
             else if (dt->shared->u.vlen.type == H5T_VLEN_STRING) {
                 /* Free the VL string */
+                char *s;
+
+                /* memcpy handles misaligned memory access safely */
+                H5MM_memcpy(&s, elem, sizeof(char *));
+
                 if (free_func != NULL)
-                    (*free_func)(*(char **)elem, free_info);
+                    (*free_func)(s, free_info);
                 else
-                    free(*(char **)elem);
+                    free(s);
             }
             else {
                 assert(0 && "Invalid VL type");
