diff -ruNw ./src/H5VLpassthru.c /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5VLpassthru.c
--- ./src/H5VLpassthru.c	2025-10-21 23:54:56.534967100 +0000
+++ /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5VLpassthru.c	2025-10-29 16:21:55.103743500 +0000
@@ -41,6 +41,11 @@
 /* This connector's header */
 #include "H5VLpassthru.h"
 
+/* For Rprintf */
+#define R_NO_REMAP 1
+#include <R_ext/Print.h>
+
+
 /**********/
 /* Macros */
 /**********/
