diff -ruNw ./src/H5Shyper.c /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5Shyper.c
--- ./src/H5Shyper.c	2025-10-21 23:54:54.840950900 +0000
+++ /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5Shyper.c	2025-10-22 21:29:26.029201300 +0000
@@ -12330,7 +12330,7 @@
     if (diminfo->start >= clip_size) {
         ret_value = 0;
         if (partial)
-            partial = false;
+            *partial = false; /* Corrected assignment to pointer. */
     } /* end if */
     else {
         /* Calculate index of first incomplete block */
