diff -ruNw ./hl/src/H5LTanalyze.c /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/hl/src/H5LTanalyze.c
--- ./hl/src/H5LTanalyze.c	2025-10-21 23:54:41.468947000 +0000
+++ /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/hl/src/H5LTanalyze.c	2025-10-29 15:51:11.779479000 +0000
@@ -636,7 +636,7 @@
 static yy_state_type yy_get_previous_state ( void );
 static yy_state_type yy_try_NUL_trans ( yy_state_type current_state  );
 static int yy_get_next_buffer ( void );
-static void yynoreturn yy_fatal_error ( const char* msg  );
+static void yy_fatal_error ( const char* msg  );
 
 /* Done after the current pattern has been matched and before the
  * corresponding action - sets up yytext.
@@ -2356,7 +2356,7 @@
 #define YY_EXIT_FAILURE 2
 #endif
 
-static void yynoreturn yy_fatal_error (const char* msg )
+static void yy_fatal_error (const char* msg )
 {
 			fprintf( stderr, "%s\n", msg );
 	exit( YY_EXIT_FAILURE );
diff -ruNw ./src/H5E.c /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5E.c
--- ./src/H5E.c	2025-10-21 23:54:49.156653600 +0000
+++ /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5E.c	2025-10-29 15:50:10.850566700 +0000
@@ -517,7 +517,6 @@
  *
  *-------------------------------------------------------------------------
  */
-H5_ATTR_FORMAT(printf, 8, 9)
 herr_t
 H5Epush2(hid_t err_stack, const char *file, const char *func, unsigned line, hid_t cls_id, hid_t maj_id,
          hid_t min_id, const char *fmt, ...)
diff -ruNw ./src/H5Eprivate.h /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5Eprivate.h
--- ./src/H5Eprivate.h	2025-10-21 23:54:49.520619400 +0000
+++ /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5Eprivate.h	2025-10-29 15:50:10.849563900 +0000
@@ -200,7 +200,7 @@
 /* Library-private functions defined in H5E package */
 H5_DLL herr_t H5E_init(void);
 H5_DLL herr_t H5E_printf_stack(const char *file, const char *func, unsigned line, hid_t maj_idx,
-                               hid_t min_idx, const char *fmt, ...) H5_ATTR_FORMAT(printf, 6, 7);
+                               hid_t min_idx, const char *fmt, ...);
 H5_DLL herr_t H5E_clear_stack(void);
 H5_DLL herr_t H5E_dump_api_stack(void);
 H5_DLL void   H5E_pause_stack(void);
diff -ruNw ./src/H5RS.c /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5RS.c
--- ./src/H5RS.c	2025-10-21 23:54:54.682485500 +0000
+++ /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5RS.c	2025-10-29 15:50:10.849563900 +0000
@@ -352,7 +352,6 @@
  *      allows diagnostic pragmas to be toggled outside of functions.
  */
 H5_GCC_CLANG_DIAG_OFF("format-nonliteral")
-H5_ATTR_FORMAT(printf, 2, 3)
 herr_t
 H5RS_asprintf_cat(H5RS_str_t *rs, const char *fmt, ...)
 {
diff -ruNw ./src/H5RSprivate.h /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5RSprivate.h
--- ./src/H5RSprivate.h	2025-10-21 23:54:54.698283800 +0000
+++ /d/Dropbox/Baylor/GitHub/hdf5lib/src/hdf5-1.14.6/src/H5RSprivate.h	2025-10-29 15:50:26.299432200 +0000
@@ -44,7 +44,7 @@
 H5_DLL herr_t      H5RS_decr(H5RS_str_t *rs);
 H5_DLL herr_t      H5RS_incr(H5RS_str_t *rs);
 H5_DLL H5RS_str_t *H5RS_dup(H5RS_str_t *s);
-H5_DLL herr_t      H5RS_asprintf_cat(H5RS_str_t *rs, const char *fmt, ...) H5_ATTR_FORMAT(printf, 2, 3);
+H5_DLL herr_t      H5RS_asprintf_cat(H5RS_str_t *rs, const char *fmt, ...);
 H5_DLL herr_t      H5RS_acat(H5RS_str_t *rs, const char *s);
 H5_DLL herr_t      H5RS_ancat(H5RS_str_t *rs, const char *s, size_t len);
 H5_DLL herr_t      H5RS_aputc(H5RS_str_t *rs, int c);
