--- a/src/H5Iint.c
+++ b/src/H5Iint.c
@@ -385,8 +385,9 @@
         else {
             /* Check for a 'free' function and call it, if it exists */
             if (udata->type_info->cls->free_func &&
-                (udata->type_info->cls->free_func)(info->u.object, H5_REQUEST_NULL) < 0) {
+                ((herr_t(*)(void *))(udata->type_info->cls->free_func))(info->u.object) < 0) {
                 if (udata->force) {
                     /* Indicate node should be removed from list */
                     mark = true;
@@ -967,8 +968,8 @@
         /* Get the ID's type */
         type_info = H5I_type_info_array_g[H5I_TYPE(id)];
 
-        if (!type_info->cls->free_func || (type_info->cls->free_func)(info->u.object, request) >= 0) {
+        if (!type_info->cls->free_func ||
+            ((herr_t(*)(void *, void **))(type_info->cls->free_func))(info->u.object, request) >= 0) {
             /* Remove the node from the type */
             if (NULL == H5I__remove_common(type_info, id))
                 HGOTO_ERROR(H5E_ID, H5E_CANTDELETE, (-1), "can't remove ID node");