--- src/H5Dint.c.orig	2025-02-05 14:11:32.000000000 +0000
+++ src/H5Dint.c	2025-11-19 23:58:01.302267800 +0000
@@ -84,7 +84,7 @@
 static herr_t H5D__open_oid(H5D_t *dataset, hid_t dapl_id);
 static herr_t H5D__init_storage(H5D_t *dset, bool full_overwrite, hsize_t old_dim[]);
 static herr_t H5D__append_flush_setup(H5D_t *dset, hid_t dapl_id);
-static herr_t H5D__close_cb(H5VL_object_t *dset_vol_obj, void **request);
+static herr_t H5D__close_cb(void *dset_vol_obj, void **request);
 static herr_t H5D__use_minimized_dset_headers(H5F_t *file, bool *minimize);
 static herr_t H5D__prepare_minimized_oh(H5F_t *file, H5D_t *dset, H5O_loc_t *oloc);
 static size_t H5D__calculate_minimum_header_size(H5F_t *file, H5D_t *dset, H5O_t *ohdr);
@@ -285,8 +285,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5D__close_cb(H5VL_object_t *dset_vol_obj, void **request)
+H5D__close_cb(void *_dset_vol_obj, void **request)
 {
+    H5VL_object_t *dset_vol_obj = (H5VL_object_t *)_dset_vol_obj;
     herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_PACKAGE
--- src/H5T.c.orig	2025-02-05 14:11:32.000000000 +0000
+++ src/H5T.c	2025-11-19 23:59:00.749401600 +0000
@@ -387,7 +387,7 @@
 static herr_t H5T__register(H5T_pers_t pers, const char *name, H5T_t *src, H5T_t *dst, H5T_conv_func_t *conv);
 static htri_t H5T__compiler_conv(H5T_t *src, H5T_t *dst);
 static herr_t H5T__set_size(H5T_t *dt, size_t size);
-static herr_t H5T__close_cb(H5T_t *dt, void **request);
+static herr_t H5T__close_cb(void *dt, void **request);
 static herr_t H5T__init_path_table(void);
 static bool   H5T__path_table_search(const H5T_t *src, const H5T_t *dst, int *idx, int *last_cmp);
 static H5T_path_t *H5T__path_find_real(const H5T_t *src, const H5T_t *dst, const char *name,
@@ -1864,8 +1864,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5T__close_cb(H5T_t *dt, void **request)
+H5T__close_cb(void *_dt, void **request)
 {
+    H5T_t *dt = (H5T_t *)_dt;
     herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_PACKAGE
--- src/H5Fint.c.orig	2025-02-05 14:11:32.000000000 +0000
+++ src/H5Fint.c	2025-11-20 01:33:08.647821000 +0000
@@ -71,7 +71,7 @@
 /* Local Prototypes */
 /********************/
 
-static herr_t H5F__close_cb(H5VL_object_t *file_vol_obj, void **request);
+static herr_t H5F__close_cb(void *file_vol_obj, void **request);
 static herr_t H5F__set_vol_conn(H5F_t *file);
 static herr_t H5F__get_objects(const H5F_t *f, unsigned types, size_t max_index, hid_t *obj_id_list,
                                bool app_ref, size_t *obj_id_count_ptr);
@@ -204,8 +204,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5F__close_cb(H5VL_object_t *file_vol_obj, void **request)
+H5F__close_cb(void *_file_vol_obj, void **request)
 {
+	H5VL_object_t *file_vol_obj = (H5VL_object_t *)_file_vol_obj;
     herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_PACKAGE
--- src/H5FD.c.orig	2025-02-05 14:11:32.000000000 +0000
+++ src/H5FD.c	2025-11-20 01:34:06.935592500 +0000
@@ -52,7 +52,7 @@
 /********************/
 /* Local Prototypes */
 /********************/
-static herr_t H5FD__free_cls(H5FD_class_t *cls, void **request);
+static herr_t H5FD__free_cls(void *cls, void **request);
 static herr_t H5FD__query(const H5FD_t *f, unsigned long *flags /*out*/);
 
 /*********************/
@@ -161,8 +161,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5FD__free_cls(H5FD_class_t *cls, void H5_ATTR_UNUSED **request)
+H5FD__free_cls(void *_cls, void H5_ATTR_UNUSED **request)
 {
+	H5FD_class_t *cls = (H5FD_class_t *)_cls;
     herr_t ret_value = SUCCEED;
 
     FUNC_ENTER_PACKAGE
--- src/H5Gint.c.orig	2025-11-20 18:29:10.588646400 +0000
+++ src/H5Gint.c	2025-11-20 18:57:19.924023400 +0000
@@ -83,7 +83,7 @@
 
 static herr_t H5G__open_oid(H5G_t *grp);
 static herr_t H5G__visit_cb(const H5O_link_t *lnk, void *_udata);
-static herr_t H5G__close_cb(H5VL_object_t *grp_vol_obj, void **request);
+static herr_t H5G__close_cb(void *grp_vol_obj, void **request);
 
 /*********************/
 /* Package Variables */
@@ -202,8 +202,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5G__close_cb(H5VL_object_t *grp_vol_obj, void **request)
+H5G__close_cb(void *_grp_vol_obj, void **request)
 {
+    H5VL_object_t *grp_vol_obj = (H5VL_object_t *) _grp_vol_obj;
     herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_PACKAGE
--- src/H5Aint.c.orig	2025-11-20 18:25:11.505964100 +0000
+++ src/H5Aint.c	2025-11-20 18:42:43.661276900 +0000
@@ -72,7 +72,7 @@
 /* Local Prototypes */
 /********************/
 
-static herr_t H5A__close_cb(H5VL_object_t *attr_vol_obj, void **request);
+static herr_t H5A__close_cb(void *attr_vol_obj, void **request);
 static herr_t H5A__compact_build_table_cb(H5O_t *oh, H5O_mesg_t *mesg /*in,out*/, unsigned sequence,
                                           unsigned *oh_flags_ptr, void *_udata /*in,out*/);
 static herr_t H5A__dense_build_table_cb(const H5A_t *attr, void *_udata);
@@ -1231,8 +1231,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5A__close_cb(H5VL_object_t *attr_vol_obj, void **request)
+H5A__close_cb(void *_attr_vol_obj, void **request)
 {
+    H5VL_object_t *attr_vol_obj = (H5VL_object_t *)_attr_vol_obj;
     herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_PACKAGE
--- src/H5VLint.c.orig	2025-11-20 18:50:02.417641400 +0000
+++ src/H5VLint.c	2025-11-20 18:52:01.599889100 +0000
@@ -92,7 +92,7 @@
 /********************/
 /* Local Prototypes */
 /********************/
-static herr_t         H5VL__free_cls(H5VL_class_t *cls, void **request);
+static herr_t         H5VL__free_cls(void *cls, void **request);
 static int            H5VL__get_connector_cb(void *obj, hid_t id, void *_op_data);
 static void          *H5VL__wrap_obj(void *obj, H5I_type_t obj_type);
 static H5VL_object_t *H5VL__new_vol_obj(H5I_type_t type, void *object, H5VL_t *vol_connector, bool wrap_obj);
@@ -284,8 +284,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5VL__free_cls(H5VL_class_t *cls, void H5_ATTR_UNUSED **request)
+H5VL__free_cls(void *_cls, void H5_ATTR_UNUSED **request)
 {
+    H5VL_class_t *cls = (H5VL_class_t *) _cls;
     herr_t ret_value = SUCCEED;
 
     FUNC_ENTER_PACKAGE
