--- src/H5Dint.c.orig	2025-02-05 14:11:32.000000000 +0000
+++ src/H5Dint.c	2025-11-19 23:58:01.302267800 +0000
@@ -84,7 +84,7 @@
 static herr_t H5D__open_oid(H5D_t *dataset, hid_t dapl_id);
 static herr_t H5D__init_storage(H5D_t *dset, bool full_overwrite, hsize_t old_dim[]);
 static herr_t H5D__append_flush_setup(H5D_t *dset, hid_t dapl_id);
-static herr_t H5D__close_cb(H5VL_object_t *dset_vol_obj, void **request);
+static herr_t H5D__close_cb(void *dset_vol_obj, void **request);
 static herr_t H5D__use_minimized_dset_headers(H5F_t *file, bool *minimize);
 static herr_t H5D__prepare_minimized_oh(H5F_t *file, H5D_t *dset, H5O_loc_t *oloc);
 static size_t H5D__calculate_minimum_header_size(H5F_t *file, H5D_t *dset, H5O_t *ohdr);
@@ -285,8 +285,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5D__close_cb(H5VL_object_t *dset_vol_obj, void **request)
+H5D__close_cb(void *_dset_vol_obj, void **request)
 {
+    H5VL_object_t *dset_vol_obj = (H5VL_object_t *)_dset_vol_obj;
     herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_PACKAGE
--- src/H5T.c.orig	2025-02-05 14:11:32.000000000 +0000
+++ src/H5T.c	2025-11-19 23:59:00.749401600 +0000
@@ -387,7 +387,7 @@
 static herr_t H5T__register(H5T_pers_t pers, const char *name, H5T_t *src, H5T_t *dst, H5T_conv_func_t *conv);
 static htri_t H5T__compiler_conv(H5T_t *src, H5T_t *dst);
 static herr_t H5T__set_size(H5T_t *dt, size_t size);
-static herr_t H5T__close_cb(H5T_t *dt, void **request);
+static herr_t H5T__close_cb(void *dt, void **request);
 static herr_t H5T__init_path_table(void);
 static bool   H5T__path_table_search(const H5T_t *src, const H5T_t *dst, int *idx, int *last_cmp);
 static H5T_path_t *H5T__path_find_real(const H5T_t *src, const H5T_t *dst, const char *name,
@@ -1864,8 +1864,9 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5T__close_cb(H5T_t *dt, void **request)
+H5T__close_cb(void *_dt, void **request)
 {
+    H5T_t *dt = (H5T_t *)_dt;
     herr_t ret_value = SUCCEED; /* Return value */
 
     FUNC_ENTER_PACKAGE
